ui/src/components/microsegmentation/RuleTable.js

```diff
@@ -0,0 +1,208 @@
+            let color = '';
+            if (i % 2 === 0) {
+                color = colors.row;
+            }
+            return (
+                <RuleRow
+                    category={this.props.category}
+                    domain={domain}
+                    details={item}
+                    idx={i}
+                    color={color}
+                    api={this.api}
+                    key={item.source_service + item.source_port}
+                    onUpdateSuccess={this.props.onSubmit}
+                    _csrf={this.props._csrf}
+                />
+            );
+        });
+
+        if (!this.state.expanded) {
+            return (
+                <StyleTable data-testid='segmentation-rule-table'>
+                    <thead>
+                        <TableCaptionStyled>
+                            <LeftMarginSpan>
+                                <Icon
+                                    icon={
+                                        this.state.expanded
+                                            ? arrowup
+                                            : arrowdown
+                                    }
+                                    onClick={expandRules}
+                                    color={colors.icons}
+                                    isLink
+                                    size={'1.25em'}
+                                    verticalAlign={'text-bottom'}
+                                />
+                            </LeftMarginSpan>
+                            {`${caption} (${length})`}
+                        </TableCaptionStyled>
+                    </thead>
+                </StyleTable>
+            );
+        }
+
+        return (
+            <StyleTable data-testid='segmentation-rule-table'>
+                <colgroup>
+                    <col style={{ width: 20 + '%' }} />
+                    <col style={{ width: 20 + '%' }} />
+                    <col style={{ width: 20 + '%' }} />
+                    <col style={{ width: 20 + '%' }} />
+                    <col style={{ width: 10 + '%' }} />
+                    <col style={{ width: 10 + '%' }} />
+                </colgroup>
+                <thead>
+                    <TableCaptionStyled>
+                        <LeftMarginSpan>
+                            <Icon
+                                icon={this.state.expanded ? arrowup : arrowdown}
+                                onClick={expandRules}
+                                color={colors.icons}
+                                isLink
+                                size={'1.25em'}
+                                verticalAlign={'text-bottom'}
+                            />
+                        </LeftMarginSpan>
+                        {`${caption} (${length})`}
+                    </TableCaptionStyled>
+                    <tr>
+                        <TableHeadStyledRoleName align={left}>
+                            {this.props.category === 'inbound'
```

Comment:
move this to a variable and use it everywhere instead of using if at all the places below